buildSrc/src/main/groovy/org/gradle/build/docs/CacheableAsciidoctorTask.groovy

```diff
@@ -15,33 +15,129 @@
  */
 package org.gradle.build.docs
 
+import org.asciidoctor.gradle.AsciidoctorProxy
 import org.asciidoctor.gradle.AsciidoctorTask
+import org.asciidoctor.gradle.ResourceCopyProxy
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.file.CopySpec
 import org.gradle.api.file.FileCollection
 import org.gradle.api.file.FileTree
 import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Classpath
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFiles
 import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.Optional
 import org.gradle.api.tasks.PathSensitive
 import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.util.PatternSet
 
 @CacheableTask
 class CacheableAsciidoctorTask extends AsciidoctorTask {
 
+    @Internal
+    @Override
+    AsciidoctorProxy getAsciidoctor() {
+        return super.getAsciidoctor()
+    }
+
+    @Internal
+    @Override
+    List<Object> getAsciidoctorExtensions() {
+        return super.getAsciidoctorExtensions()
+    }
+
+    @PathSensitive(PathSensitivity.RELATIVE)
+    @Override
+    File getBaseDir() {
+        return super.getBaseDir()
+    }
+
+    @Classpath
+    @Optional
+    @Override
+    Configuration getClasspath() {
```

Comment:
The task action does the following:
```
if (classpath == null) {
    classpath = project.configurations.getByName(AsciidoctorPlugin.ASCIIDOCTOR)
}
```

That is not very nice :(

Should this method return that instead:
```
super.getClasspath() == null ? project.configurations.getByName(AsciidoctorPlugin.ASCIIDOCTOR) : super.getClasspath()
```

Reply:
There are so many things to fix about this task, I don't think it makes sense to fix this single problem. The goal of `CacheableAsciidoctorTask` is to make it work for us in any case. We should instead send a PR to Asciidoctor with fixes.

Reply:
👍